tests/components/input_text/test_init.py

```diff
@@ -81,16 +81,21 @@ async def async_set_value(hass: HomeAssistant, entity_id: str, value: str) -> No
     )
 
 
-async def test_config(hass: HomeAssistant) -> None:
-    """Test config."""
-    invalid_configs = [
+@pytest.mark.parametrize(
+    "invalid_config",
+    [
         None,
-        {},
         {"name with space": None},
-        {"test_1": {"min": 50, "max": 50}},
-    ]
```

Comment:
Looking at this test, it seems it has never worked properly.

If async_setup_component fails, calling it again with any other different config just keeps returning the original failure. So no matter what you listed in invalid_configs after the first `None`, the test will always pass, even with valid configs. 

I don't know if there's any way to make this work within a loop, but using parameterize seems to maintain the intent of the test.

It seems like all of the old-style helpers have the some variant of this broken test, so we should probably go fix that after this, if this is a good solution.

Reply:
Both `{}` and `{"test_1": {"min": 50, "max": 50}}` are actually valid configs.

Reply:
It indeed looks like we should fix the other tests which might also be looping like this

Reply:
https://github.com/home-assistant/core/pull/148965